src/ResourceManager/Profile/Commands.Profile.Test/ArgumentCompleterTests.ps1

```diff
@@ -39,4 +39,53 @@ function Test-ResourceGroupCompleter
 	$resourceGroups = [Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters.ResourceGroupCompleterAttribute]::GetResourceGroups(-1)
 	$expectResourceGroups = Get-AzureRmResourceGroup | ForEach-Object {$_.ResourceGroupName}
 	Assert-AreEqualArray $resourceGroups $expectResourceGroups
+}
+
+<#
+.SYNOPSIS
+Tests resource name completer
+#>
+function Test-ResourceNameCompleter
+{
+    $filePath = Join-Path -Path $PSScriptRoot -ChildPath "\Microsoft.Azure.Commands.ResourceManager.Common.dll"
+    [System.Reflection.Assembly]::LoadFrom($filePath)
+    $resourceType = "Microsoft.Storage/storageAccounts"
+    $expectResourceNames = Get-AzureRmResource -ResourceType  $resourceType | ForEach-Object {$_.Name}
+    # take data from Azure and put to cache
+    [Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters.ResourceNameCompleterAttribute]::new($resourceType, "ResourceGroupName", $Null)
+    $resourceNames = [Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters.ResourceNameCompleterAttribute]::GetResources($resourceType) | ForEach-Object { $_.Name }
+    Assert-AreEqualArray $resourceNames $expectResourceNames
+    # take data from the cache
+    $resourceNames = [Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters.ResourceNameCompleterAttribute]::GetResources($resourceType) | ForEach-Object { $_.Name }
+    Assert-AreEqualArray $resourceNames $expectResourceNames
+    # change time to update the cache
+    [Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters.ResourceNameCompleterAttribute]::TimeToUpdate = [System.TimeSpan]::FromSeconds(1)
+    Start-Sleep -Seconds 1
```

Comment:
Use wait-seconds: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ps1#L307. We don't allow sleeps in tests.

Reply:
fixed